src/pretix/presale/views/customer.py

```diff
@@ -541,3 +568,286 @@ def get(self, request, *args, **kwargs):
+                    self.provider,
+                    request.GET.get('code'),
+                    redirect_uri,
+                )
+            except ValidationError as e:
+                for msg in e:
+                    return self._fail(msg, popup_origin)
+        else:
+            raise Http404("Unknown SSO method.")
+
+        identifier = hashlib.sha256(
+            profile['uid'].encode() + b'@' + str(self.provider.pk).encode()
+        ).hexdigest().upper()[:settings.ENTROPY['customer_identifier']]
+        if "1" not in identifier and "0" not in identifier:
+            # This is a hack to make sure the hash space does not overlap with the random identifiers generated by
+            # Customer.assign_identifier()
+            identifier = identifier[:4] + "1" + identifier[4:-1]
+
+        try:
+            customer = self.request.organizer.customers.get(
+                provider=self.provider,
+                identifier=identifier,
+            )
+        except Customer.MultipleObjectsReturned:
+            return self._fail(
+                _('Login was not successful. Error message: "{error}".').format(
+                    error='identifier not unique',
+                ),
+                popup_origin,
+            )
+        except Customer.DoesNotExist:
+            name_scheme = self.request.organizer.settings.name_scheme
+            name_parts = {
+                '_scheme': name_scheme,
+            }
+            scheme = PERSON_NAME_SCHEMES.get(name_scheme)
+            for fname, label, size in scheme['fields']:
+                if fname in profile:
+                    name_parts[fname] = profile[fname]
+            if len(name_parts) == 1 and profile.get('name'):
+                name_parts = {'_legacy': profile['name']}
+            customer = Customer(
+                organizer=self.request.organizer,
+                identifier=identifier,
+                external_identifier=profile['uid'],
+                provider=self.provider,
+                email=profile['email'],
+                phone=profile.get('phone') or None,
+                name_parts=name_parts,
+                is_active=True,
+                is_verified=True,  # todo: always?
+                locale=request.LANGUAGE_CODE,
+            )
+            try:
+                customer.save(force_insert=True)
+            except IntegrityError:
+                # This might either be a race condition or the email address is taken
+                # by a different customer account
+                try:
+                    customer = self.request.organizer.customers.get(
+                        provider=self.provider,
+                        identifier=identifier,
+                    )
+                except Customer.DoesNotExist:
+                    return self._fail(
+                        _('We were unable to use your login since the email address is already used for a different '
```

Comment:
Should we output the email in this error message? I know this is a rare case, but I authorized pretix as a SSO-client in my Google account in a previous, failed attempt to login and now, when I press the „Login with Google“ Button I immediately get redirected from Google back to the login page with this error message. I need to go to the google homepage to see with what account/email I am logged in.